### ISSUE TITLE
Provide ability to tap both system out and err at the same time

### ISSUE THREAD

**Issue by rnc** [stefanbirkner/system-lambda#8]
This would be useful for executing CLI type applications that output to both out/err and you wish to assert on the entire 'console type' output.

**Comment by stefanbirkner** [comment]
I started working on it.

**Comment by stefanbirkner** [comment]
What do you consider a better name: `tapSystemErrAndOut` or `tapConsole`?

**Comment by rnc** [comment]
I think the former ; `tapSystemErrAndOut` makes it more explicit and follows the same naming pattern as the others.

**Comment by ashleyfrieze** [comment]
I think this is always going to be challenging. You have the choice of nesting constructs:

```java
withThing1(() -> {
   withThing2(() -> {
   });
});
```

or you can start to create the various permutations of setup function:

```java
withThing1AndThing2(() -> {
});
```

But the problem is that you can never quite predict all the permutations... and the nesting of the execute-around idiom gets old quickly.

This solution would work better if you could **compose** the system lambdas.

One possibility is to create a `compose` function. This might be something like this:

```java
@FunctionalInterface
public interface ThrowingFunction<T> {
    T apply(Callable<T> callable) throws Exception;

}

public static <T> T compose(Callable<T> codeUnderTest, ThrowingFunction<T> ... executeAround);
```

This might be used like this:

```java
T testValue = compose(() -> callCodeUnderTest,
    callable -> withEnvironmentVariable("a", "b").execute(callable),
    SystemLambda::withMuteSystemErr,
    SystemLambda::withMuteSystemOut);
```

However... the structure of SystemLambda still fights this as the facade functions are doing one job in a very opaque way and so you can't easily compose multiple ones.

**Comment by stefanbirkner** [comment]
The feature is implemented in System Lambda 1.2.0: a203454212b4317b111c58aeab2ec72a6f8b6690
